fhir-bulkimportexport/src/main/java/com/ibm/watson/health/fhir/bulkexport/BulkExportBatchLet.java

```diff
@@ -149,250 +160,230 @@
-            log("process", "Set tenant to default!");
+            logger.info("process: Set tenant to default!");
         }
-        
         if (fhirDatastoreId == null) {
             fhirDatastoreId = Constants.DEFAULT_FHIR_TENANT;
-            log("readItem", "Set DatastoreId to default!");
+            logger.info("process: Set DatastoreId to default!");
         }
-        
+        if (fhirSearchPageSize != null) {
+            try {
+                pageSize = Integer.parseInt(fhirSearchPageSize);
+                logger.info("process: Set page size to " + pageSize + ".");
+            } catch (Exception e) {
+                logger.warning("process: Set page size to default(" + Constants.DEFAULT_SEARCH_PAGE_SIZE + ").");
+            }
+        }
+        if (!isSingleCosObject && cosBucketMaxFileSize != null) {
+            try {
+                cosMaxFileSize = Integer.parseInt(cosBucketMaxFileSize);
+                logger.info("process: Set max COS file size to " + cosMaxFileSize + ".");
+            } catch (Exception e) {
+                cosMaxFileSize = Constants.DEFAULT_MAXCOSFILE_SIZE;
+                logger.warning("process: Set max COS file size to default(" + Constants.DEFAULT_MAXCOSFILE_SIZE + ").");
+            }
+        }
+
         FHIRConfiguration.setConfigHome("./");
         FHIRRequestContext.set(new FHIRRequestContext(fhirTenant, fhirDatastoreId));
-
         FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper();
         FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
-
         Class<? extends Resource> resourceType = (Class<? extends Resource>) ModelSupport
                 .getResourceType(fhirResourceType);
         FHIRSearchContext searchContext;
         FHIRPersistenceContext persistenceContext;
         Map<String, List<String>> queryParameters = new HashMap<>();
-        String queryString;
+        String queryString = "&_sort=_lastUpdated";
 
-        queryString = "&_lastUpdated=ge" + fhirSearchFromDate + "&_lastUpdated=lt" + fhirSearchToDate
-                + "&_sort=_lastUpdated";
-        queryParameters.put("_lastUpdated", Collections.singletonList("ge" + fhirSearchFromDate));
-        queryParameters.put("_lastUpdated", Collections.singletonList("lt" + fhirSearchToDate));
-        queryParameters.put("_sort", Arrays.asList(new String[] { "_lastUpdated" }));
+        if (fhirSearchFromDate != null) {
+            queryString += ("&_lastUpdated=ge" + fhirSearchFromDate);
+            queryParameters.put("_lastUpdated", Collections.singletonList("ge" + fhirSearchFromDate));
+        }
+        if (fhirSearchToDate != null) {
+            queryString += ("&_lastUpdated=lt" + fhirSearchToDate);
+            queryParameters.put("_lastUpdated", Collections.singletonList("lt" + fhirSearchToDate));
+        }
 
+        queryParameters.put("_sort", Arrays.asList(new String[] { "_lastUpdated" }));
```

Comment:
is queryParameters Object[] ? or String?  couldn't you use `Arrays.asList()'. 
Couldn't you make the `_lastUpdated` a constant?

Reply:
This is mimic of the test codes of search project, queryParameters is a hashmap. 
yeah, it's good idea to make _lastUpdated a constant.